src/ngAnimate/animateQueue.js

```diff
@@ -579,6 +586,11 @@ var $$AnimateQueueProvider = ['$animateProvider', function($animateProvider) {
       var animateChildren;
       var elementDisabled = disabledElementsLookup.get(getDomNode(element));
 
+      if (elementDisabled === true) {
+        // If animations on the element have been disabled explicitly, no need to traverse the parents
+        return false;
+      }
```

Comment:
But line 626 is inside the loop, and this check comes before the loop. Also, this is checked before areAnimationsAllowed, too. The lookup here was added later to catch elements that were explicitly allowed.